fdbclient/include/fdbclient/BulkLoading.h

```diff
@@ -25,63 +25,290 @@
+	                 BulkLoadType loadType,
+	                 BulkLoadTransportMethod transportMethod)
+	  : fileSet(fileSet), beginKey(beginKey), endKey(endKey), version(version), checksum(checksum), bytes(bytes),
+	    byteSampleSetting(byteSampleSetting), loadType(loadType), transportMethod(transportMethod) {
+		ASSERT(isValid());
+	}
+
+	// Used when initialize a bulk dump job. Information are partially filled at this time.
+	BulkLoadManifest(BulkLoadType loadType, BulkLoadTransportMethod transportMethod, const std::string& rootPath)
+	  : loadType(loadType), transportMethod(transportMethod) {
+		fileSet = BulkLoadFileSet(rootPath);
+	}
+
+	bool isValid() const {
+		if (beginKey >= endKey) {
+			return false;
+		}
+		if (!fileSet.isValid()) {
+			return false;
+		}
+		if (!byteSampleSetting.isValid()) {
+			return false;
+		}
+		if (transportMethod == BulkLoadTransportMethod::Invalid) {
+			return false;
+		} else if (transportMethod != BulkLoadTransportMethod::CP &&
+		           transportMethod != BulkLoadTransportMethod::BLOBSTORE) {
+			ASSERT(false);
+		}
+		if (loadType == BulkLoadType::Invalid) {
+			return false;
+		} else if (loadType != BulkLoadType::SST) {
+			ASSERT(false);
+		}
+		return true;
+	}
+
+	std::string getRootPath() const { return fileSet.rootPath; }
+
+	KeyRange getRange() const { return Standalone(KeyRangeRef(beginKey, endKey)); }
+
+	BulkLoadTransportMethod getTransportMethod() const { return transportMethod; }
+
+	BulkLoadType getLoadType() const { return loadType; }
+
+	std::string getBeginKeyString() const { return beginKey.toFullHexStringPlain(); }
+
+	std::string getEndKeyString() const { return endKey.toFullHexStringPlain(); }
+
+	void setRange(const KeyRange& range) {
+		ASSERT(!range.empty() && beginKey.empty() && endKey.empty());
+		beginKey = range.begin;
+		endKey = range.end;
+		return;
+	}
+
+	// Generating human readable string to stored in the manifest file
+	std::string toString() const {
+		return fileSet.toString() + ", [BeginKey]: " + getBeginKeyString() + ", [EndKey]: " + getEndKeyString() +
+		       ", [Version]: " + std::to_string(version) + ", [Checksum]: " + checksum +
+		       ", [Bytes]: " + std::to_string(bytes) + ", " + byteSampleSetting.toString() +
+		       ", [loadType]: " + std::to_string(static_cast<uint8_t>(loadType)) +
+		       ", [TransportMethod]: " + std::to_string(static_cast<uint8_t>(transportMethod));
+	}
+
+	// Generate human readable string as an entry in the job manifest file.
+	// A job is partitioned into tasks by ranges.
+	// The job manifest includes all manifest files for all ranges within the job.
+	std::string generateEntryInJobManifest() const {
+		return getBeginKeyString() + ", " + getEndKeyString() + ", " + std::to_string(version) + ", " +
+		       std::to_string(bytes) + ", " + joinPath(fileSet.relativePath, fileSet.manifestFileName);
+	}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
```

Comment:
Should this be versioned?

Reply:
Why do we want to do the version at this point? Thanks!